govc/session/login.go

```diff
@@ -149,16 +147,30 @@ func (cmd *login) issueToken(ctx context.Context, vc *vim25.Client) (string, err
 	req := sts.TokenRequest{
 		Certificate: c.Certificate(),
 		Userinfo:    cmd.Userinfo(),
-		Delegatable: true,
-		ActAs:       cmd.actas,
+		Renewable:   true,
+		Token:       cmd.token,
 		Lifetime:    cmd.life,
 	}
 
+	if req.Certificate == nil {
+		req.Delegatable = true // Bearer token request
+	}
+
 	s, err := c.Issue(ctx, req)
 	if err != nil {
 		return "", err
 	}
 
+	duration := s.Lifetime.Expires.Sub(s.Lifetime.Created)
```

Comment:
Hi @dougm,

For some reason my previous comment here did not get added to my review. I wondered if the following variation might fall in line more with your coding style and at the same time keep `duration` out of the scope of the function:

```go
if req.Token != "" {
 	duration := s.Lifetime.Expires.Sub(s.Lifetime.Created)
 	if duration != req.Lifetime
```